### PR TITLE
tests: add revert error strings to assertion checks

### LINKED ISSUE #816: tests: add revert error strings to assertion checks
`assertRevert()` now allows us to check expected revert strings, greatly improving our confidence in revert tests.

Payroll and Finance have already started using them, and it would be great to include in the other apps (Agent, Token Manager, Vault, and Voting).

### PR DESCRIPTION
**Pull request by haythem96**
Fixes #816

### CODE REVIEW

File: apps/agent/test/agent_shared.js [hunk 1]
```diff
@@ -40,6 +41,21 @@ module.exports = (
...
+      // aragonOS errors
+      APP_AUTH_FAILED: 'APP_AUTH_FAILED',
+      INIT_ALREADY_INITIALIZED: 'INIT_ALREADY_INITIALIZED',
+      INIT_NOT_INITIALIZED: 'INIT_NOT_INITIALIZED',
+      RECOVER_DISALLOWED: 'RECOVER_DISALLOWED',
+
+      // Agent errors
+      AGENT_EXEC_ETH_NO_DATA = "AGENT_EXEC_ETH_NO_DATA",
+      AGENT_EXEC_TARGET_NO_CONTRACT = "AGENT_EXEC_TARGET_NO_CONTRACT",
+      AGENT_DESIGNATED_TO_SELF = "AGENT_DESIGNATED_TO_SELF",
```
Comment (sohkai) [inline apps/agent/test/agent_shared.js]: Oops, you need to use `:` here (and in some of the other tests, where you're adding keys to an object).

Comment (haythem96) [reply apps/agent/test/agent_shared.js]: arrrg yup true, copy/paster error XD

**Comment by sohkai** [comment]
@haythem96 It looks like there are still a few tests not passing due to incorrect revert strings, do you mind fixing those?

**Comment by sohkai** [comment]
@haythem96 Just wanted to ping, could you also merge with master? 🤗

**Comment by haythem96** [comment]
> @haythem96 It looks like there are still a few tests not passing due to incorrect revert strings, do you mind fixing those?

Fixed.

**Comment by haythem96** [comment]
can we merge this one ?

**Comment by sohkai** [comment]
@haythem96 I will be looking at this soon!

**Comment by sohkai** [comment]
@haythem96 I've pushed a few commits that add a bit of polish to the existing tests, especially those where we weren't checking for a revert case (now easy to detect thanks to these revert messages) :).

We'll get this merged within the next few days!

**Comment by haythem96** [comment]
> @haythem96 I've pushed a few commits that add a bit of polish to the existing tests, especially those where we weren't checking for a revert case (now easy to detect thanks to these revert messages) :).
> 
> We'll get this merged within the next few days!

awesome.

**Pull request merged**